File: Data Structures & Algorithms/valid-sudoku/submission-1.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        int[] rows = new int[9];
        int[] cols = new int[9];
        int[] boxes = new int[9];

        for (int row = 0; row < 9; row++) {
            for (int col = 0; col < 9; col++) {
                char cell = board[row][col];
                if (cell == '.') {
                    continue;
                }

                int val = 1 << ((cell - '0') - 1);
                if ((rows[row] & val) != 0) {
                    return false;
                }
                if ((cols[col] & val) != 0) {
                    return false;
                }
                if ((boxes[row / 3 * 3 + col / 3] & val) != 0) {
                    return false;
                }

                rows[row] |= val;
                cols[col] |= val;
                boxes[row / 3 * 3 + col / 3] |= val;
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-1.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        rows, cols = len(matrix), len(matrix[0])
        firstRowZeroed = False
        firstColumnZeroed = False

        for col in range(cols):
            if matrix[0][col] == 0:
                firstRowZeroed = True
                break

        for row in range(len(matrix)):
            if matrix[row][0] == 0:
                firstColumnZeroed = True
                break

        for row in range(1, rows):
            for col in range(1, cols):
                if matrix[row][col] == 0:
                    matrix[0][col] = 0
                    matrix[row][0] = 0

        for col in range(1, cols):
            if matrix[0][col] == 0:
                for row in range(1, rows):
                    matrix[row][col] = 0

        for row in range(rows):
            if matrix[row][0] == 0:
                for col in range(1, cols):
                    matrix[row][col] = 0

        if firstRowZeroed:
            for col in range(cols):
                matrix[0][col] = 0

        if firstColumnZeroed:
            for row in range(rows):
                matrix[row][0] = 0